.github/workflows/CI.yml

```diff
@@ -215,6 +215,271 @@ jobs:
 
       - uses: bcomnes/cleanup-xvfb@v1
 
+  build-win:
```

Comment:
I would move the Window CI into a separate workflow (as a separate file) to avoid making the "main" CI's configuration more complex

Reply:
I agree the file is very long and needs to be split... I wonder however if that is the right split: I was thinking that maybe it is better to split functionally instead of per platform. For instance have a file that is for quick checks (fmt etc) and then a file with the full build matrix that runs if the first is successfull

Reply:
There's #247 for a little clean up of the build. The CI as is right now is just bare-bone to get us started post-migration to a single repository, any improvements are very much welcome :)

Reply:
The argument still stand. :)

Reply:
I am not sure what you are asking me to do to be honest... when we started the CI file was much longer and had a lot of jobs and steps, but right now even with my patch it is 270 which is not bad. I looked into various other rust repositories and the standard practice seems to be to use different workflows when they have different triggers.